src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.html

```diff
@@ -82,6 +82,9 @@
                    class="form-control"
                    type="text"
                    formControlName="display_name">
+            <span class="invalid-feedback"
+                  *ngIf="userForm.showError('display_name', frm, 'pattern')"
```

Comment:
@Pegonzal don't take me wrong, you have done as everyone (myself included) would have done. You look at how this has been done previously, and simply repeat. And I personally find that it's better to do things consistently (yet _slightly_) wrong than randomly right-wrong. So in this very case, I'd go for merging this and if I were you then I'd go and open 2 trackers:
* Add the tslint expression that forbids methods in templates (this will involve replacing the 277 appearances of the `showError()` so probably the pipe approach would work great here since it's easy to replace with some `sed`/regex bulk commands).
* Explore improvements to the vanilla Angular validators (I always felt they were lacking... what about defining the error message in the validator itself rather than in the HTML template?). I just googled and found the following frameworks:
  * [`@rxweb/reactive-form-validators`](https://dev.to/ushmidave/improve-your-way-to-validate-angular-reactive-forms-574l): decorator based, validation error messages can be defined in a single location. Like it.
  * [Dynamic Forms](https://dev.to/myndpm/a-new-approach-to-have-dynamic-forms-in-angular-5d11): looks promising, but not sure if it works well with non-Material components.
  * [JSON Forms](https://www.npmjs.com/package/@jsonforms/angular): even if we don't want to generate the forms from the back-end, we should start exploring this approach, which heavily minimize the amount of boilerplate for form creation and maintenance.
  * Maybe you can find other, or try your way towards improving Angular forms/validators.

Reply:
Okay perfect, then I understood you perfectly, just wanted to clarify. Thanks Ernesto!